powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehaviorTest.java

```diff
@@ -89,8 +89,27 @@ void testCreateOperationWithActivationId() throws Exception {
 
         final OperationDetailResponse operationDetailResponse = operationService.createOperation(request);
         final OperationEntity savedEntity = operationRepository.findOperation(operationDetailResponse.getId()).get();
+
+        assertTrue(operationRepository.findOperation(operationDetailResponse.getId()).isPresent());
+        assertEquals(ACTIVATION_ID, savedEntity.getActivationId());
+        assertNull(operationDetailResponse.getProximityOtp());
+    }
+
+    @Test
+    void testCreateOperationWithActivationIdAndProximityCheck() throws Exception {
+        final OperationCreateRequest request = new OperationCreateRequest();
+        request.setActivationId(ACTIVATION_ID);
+        request.setApplications(List.of(APP_ID));
+        request.setTemplateName("test-template");
+        request.setUserId(USER_ID);
+        request.setProximityCheckEnabled(true);
+
+        final OperationDetailResponse operationDetailResponse = operationService.createOperation(request);
+        final OperationEntity savedEntity = operationRepository.findOperation(operationDetailResponse.getId()).get();
+
         assertTrue(operationRepository.findOperation(operationDetailResponse.getId()).isPresent());
```

Comment:
Using `operationService.operationDetail` gives us better coverage than just an operationRepository method.

Reply:
Ok, why not. It is a unit test, but `#operationDetail` is not covered yet, so it is an improvement.